### ISSUE TITLE
Japanese client support

### ISSUE THREAD

**Issue by imanakas71** [EasyFarm/EasyFarm#239]
I'd like to run this program on a Japanese client, but the spelling of items, abilities, weapon skills etc will be displayed on the client screen in English.
Is there any language mode change scheme in this program?

**Comment by Mykezero** [comment]
Sorry if I've misunderstood your question, but are you saying that the program is sending the commands to the game in English, but you'd like to send them in Japanese?

If that's the case, then I think it might be a pretty easy problem to solve ^^;

**Comment by imanakas71** [comment]
Hi, Mykezero.  You're understanding is 100% right. 
I'm a Japanese player and would like to use your program on Japanese client.
I changed Classes/AbilityParser.cs around line 162 form "english" to "japanese".
Then weaponskill worked fine, but I have no idea about job abilities etc..
I am going to re-write save data then...

**Comment by Mykezero** [comment]
I don't think there's a quick way to solve this problem: It was a bad decision for me to hard code the use of that English field. 

The code will need to be reworked so that it uses the user's specified action name, regardless of what's in the resource files. 

That user specified name is stored in the BattleAbility class which is acting as a view model / input into the executor class which sends the actions to the game. for execution. 

Besides the action names, will the action type and action target fields need adjusting as well?

`/magic "ケアル" <t>`

Would this work as input into the game client?

**Comment by imanakas71** [comment]
The resource file is full of descriptions of both Japanese and English, I believe it is very good work.

I do not think that my English is good, but will try to tell you by giving examples.
My problem is that , when "ウォークライ" is the action name /jobability "War Cry" is displayed in the window and the system does not accept that input line.
Your program seems to know that "ウォークライ" is "War Cry".  Is this because it is listed in the resource file?


All the item name, ability name, weaponskill name, magic name in Japanese are not divided by space, like
"Cure III" --> ケアルIII, so we are not allowed to put double-quote in the line like

/magic ケアル <t>

instead of

/magic "ケアル" <t>.   <--- this line does not work.

**Comment by Mykezero** [comment]
Yep. The program uses Windower's resource files which contains the spellings for actions in different languages. 

Very cool; I didn't know that the quotes would cause the program to fail. We could definitely change how quoting is applied to something like this: 

`Name.Contains(" ") ? $"\"{Name}\"" : Name`

I believe quotes are only required for actions which have multiple words in its name. A rule like that should work for both languages. 

The quoting happens [here][1] in the Ability class. 

[1]: https://github.com/EasyFarm/EasyFarm/blob/0a4bd29b1423587ffa30e2c5284810840c4d9456/EasyFarm/Parsing/Ability.cs#L173

**Comment by imanakas71** [comment]
Changed the line you mentioned like 

`public string ActionCommand => Prefix + " " + Japanese + " " + Postfix;`

abilities seems to work well ( after very small test ).
Thanks for the suggestions.  Maybe I can do something by myself if I faced to another new issues regarding the language if only you will not change these code drastically.
Anyway, many thanks.  Please close this issue.

**Comment by Mykezero** [comment]
I would definitely like to add support for both languages, and I'm glad it's working for you for now. I'm hoping to add these changes this weekend so that everyone can benefit from it. Thanks for opening this issue! ^^

**Comment by imanakas71** [comment]
You're welcome and thanks for the good codes.
Hope my contribution will not be abused :-)

**Comment by Mykezero** [comment]
I think the newest release of EasyFarm should be usable on the Japanese client now. If possible, would you mind checking it out to see if it works? Thanks for all your help!

**Comment by imanakas71** [comment]
Ok, let me check but it takes a while.

**Comment by imanakas71** [comment]
It seems working well.  
One thing I need to know for the check is about "/range" setting in battle.
For JP client, /range is just allowed to input  "/range <t>" but EasyFarm needs to add weapon section.
JP client refuges "/range (weapon) <t>".
Is it needed for other language clients?
